docs/architecture/overview.md

```diff
@@ -0,0 +1,115 @@
+# Pulumi Terraform Bridge Architecture Overview
+
+This document captures the high–level mental model maintainers use to reason about the Pulumi Terraform Bridge. It is
+intended as a launchpad: each section links to the code, docs, or playbooks that provide deeper detail.
+
+## System Goals
+
+The bridge has two equally important responsibilities:
+
+1. **Build-time generation** – Translate a Terraform provider's schema and metadata into a Pulumi schema, SDKs, and docs so Pulumi users can program against the provider.
+2. **Runtime translation** – Act as a Pulumi resource provider that speaks Pulumi's gRPC protocol while driving the
+   Terraform provider implementation under the hood.
+
+A change to either half can impact the other. Treat the bridge as one product with two tightly-coupled pipelines.
+
+## Major Components
+
+| Layer | Location | Role |
+| ----- | -------- | ---- |
+| Build-time | `pkg/tfgen`, `docs/guides/*`, `pkg/convert`, `pkg/tf2pulumi` | Introspect Terraform provider schemas, emit Pulumi schema, SDKs, and docs. |
```

Comment:
tf2pulumi needs to be removed. It's been straggling but it does not belong here.  Worth documenting accordingly so it does not get more confusing.